bio-qiime-pipeline/src/utils.py

```diff
@@ -145,3 +147,79 @@ def getMetadata(sample: CustomSample, metadataFileNme: str) -> Path:
         metadataPath = metadataPath.parent / f"{metadataPath.stem}.tsv"
 
     return metadataPath
+
+
+def getDatasetName(experiment: Experiment[CustomDataset], step: int) -> str:
+    if step < 1 or step > 5:
+        raise ValueError(">> [Microbiome analysis] Step number has to be between 1 and 5")
+
+    if experiment.parameters["barcodeColumn"]:
+        prefix = f"{experiment.id} - Step 1: Demux"
+    else:
+        prefix = f"{experiment.id} - Step 1: Import"
+
+    paramList = [
+        str(experiment.dataset.id),
+        str(experiment.parameters["metadataFileName"]),
+        str(experiment.parameters["barcodeColumn"]),
+        str(experiment.parameters["forwardAdapter"]),
+        str(experiment.parameters["reverseAdapter"])
+    ]
+
+    if step > 1:
```

Comment:
can we create a separate function for generating the name of the cached dataset for every step?